src/vos/vos_aggregate.c

```diff
@@ -442,24 +442,23 @@ static int
 csum_prepare_buf(struct agg_lgc_seg *segs, unsigned int seg_cnt,
 		 void **csum_bufp, unsigned int cur_buf, unsigned int add_len)
 {
-	void		*buffer;
-	unsigned char	*csum_buf = *csum_bufp;
+	unsigned char	*buffer;
 	unsigned int	 new_len = cur_buf + add_len;
 	int		 i;
 
 	D_ASSERT(add_len);
-	D_REALLOC(buffer, csum_buf, new_len);
+	D_REALLOC(buffer, *csum_bufp, new_len);
 	if (buffer == NULL)
 		return -DER_NOMEM;
-	csum_buf = buffer;
-	memset(&csum_buf[cur_buf], 0, add_len);
+	memset(&buffer[cur_buf], 0, add_len);
 	for (i = 0; i < seg_cnt; i++) {
 		struct dcs_csum_info *csum_info = &segs[i].ls_ent_in.ei_csum;
 
-		csum_info->cs_csum = &csum_buf[cur_buf];
+		csum_info->cs_csum = &buffer[cur_buf];
```

Comment:
I see, you used this 'cs_csum' in logical entry to pass the csum buffer pointer for recalc callback?

Reply:
Right, each logical entry in will use their calculated range in buffer for csum generation.  I the evtree_insert call, the content of this buffer is copied to PM.

The point of this JIRA is to change the pointer, following the copy, to point to the checksum array in PM.  This will allow the the allocated DRAM buffer to be freed at the end of each window.

I went ahead and did a patch just to fix the memory leak, instead of doing the change to the rest.  Will follow-up with another patch.

Reply:
The patch now completes all required changes.